addons/auth_password_pwn_checker/models/res_users.py

```diff
@@ -0,0 +1,57 @@
+import hashlib
+import logging
+from datetime import timedelta, date
+
+import requests
+
+from odoo import _, models, api, fields
+
+_logger = logging.getLogger(__name__)
+
+
+class ResUsers(models.Model):
+    _inherit = ['res.users']
+
+    pwn_last_checked = fields.Date()
+
+    @classmethod
+    def authenticate(cls, db, login, password, user_agent_env):
+        uid = super(ResUsers, cls).authenticate(db, login, password, user_agent_env)
+        cls._check_user_password(uid, password)
+        return uid
+
+    @classmethod
+    def _check_user_password(cls, uid: int, password: str):
+        with cls.pool.cursor() as cr:
+            env = api.Environment(cr, uid, {})
+            should_check = (not env.user.pwn_last_checked or
+                            env.user.pwn_last_checked < date.today() - timedelta(days=7))
+            if not should_check:
+                return
+
+            try:
+                pwned = cls._is_password_pwned(password)
+            except requests.exceptions.RequestException:
+                _logger.warning("Request to api.pwnedpasswords.com failed.")
+                return
+
+            env.user.pwn_last_checked = date.today()
+
+            if pwned:
+                odoobot_email = env.ref('base.partner_root').email_formatted
+
+                env['mail.mail'].sudo().create({
```

Comment:
Would delegate to a sub-method (split the check from the email sending itself as those are two separate functional stuff).